pkg/registry/core/rest/storage_core.go

```diff
@@ -416,43 +414,20 @@ type componentStatusStorage struct {
 	storageFactory serverstorage.StorageFactory
 }
 
-func (s componentStatusStorage) serversToValidate() map[string]*componentstatus.Server {
+func (s componentStatusStorage) serversToValidate() (map[string]componentstatus.Server, error) {
 	// this is fragile, which assumes that the default port is being used
 	// TODO: switch to secure port until these components remove the ability to serve insecurely.
-	serversToValidate := map[string]*componentstatus.Server{
-		"controller-manager": {EnableHTTPS: true, TLSConfig: &tls.Config{InsecureSkipVerify: true}, Addr: "127.0.0.1", Port: ports.KubeControllerManagerPort, Path: "/healthz"},
-		"scheduler":          {EnableHTTPS: true, TLSConfig: &tls.Config{InsecureSkipVerify: true}, Addr: "127.0.0.1", Port: kubeschedulerconfig.DefaultKubeSchedulerPort, Path: "/healthz"},
+	serversToValidate := map[string]componentstatus.Server{
+		"controller-manager": &componentstatus.HttpServer{EnableHTTPS: true, TLSConfig: &tls.Config{InsecureSkipVerify: true}, Addr: "127.0.0.1", Port: ports.KubeControllerManagerPort, Path: "/healthz"},
+		"scheduler":          &componentstatus.HttpServer{EnableHTTPS: true, TLSConfig: &tls.Config{InsecureSkipVerify: true}, Addr: "127.0.0.1", Port: kubeschedulerconfig.DefaultKubeSchedulerPort, Path: "/healthz"},
+	}
+	probers, err := s.storageFactory.Probers()
+	if err != nil {
+		return nil, err
 	}
 
-	for ix, machine := range s.storageFactory.Backends() {
```

Comment:
we already have the tlsconfig and url in the `Backends()`output, it will not be easier if you just create the etcd client here and implement the logic in this folder without having to override or modify the rest of the code?

Reply:
It might be a quicker solution, but for long term we don't need to expose url hosts as `Backends()`. Also I wanted to keep the behavior of using a resource prefix like in health probe.

Reply:
On the other hand I hit the problem of passing ownership of client lifecycle :(.